jsx/tyxml_jsx.ml

```diff
@@ -23,16 +23,22 @@ let lowercase_lead s =
   String.mapi (fun i c -> if i = 0 then Char.lowercase_ascii c else c) s
 
 let to_kebab_case name =
-  let length = String.length name in
-  if length > 5 then
-    let first = String.sub name 0 4 in
-    match first with
-    | "aria"
-    | "data" ->
-      first ^ "-" ^ lowercase_lead (String.sub name 4 (length - 4))
-    | _ -> name
-  else
-    name
+  let open Re in
+  let kebab string =
+    replace (Posix.compile_pat "[A-Z]") ~f:(fun g -> "-" ^ Group.get g 0) string
+    |> String.lowercase_ascii
+    |> replace_string (compile @@ char '_') ~by:"-" in
+  match exec_opt (Perl.compile_pat {|^(data_?|aria_?)(.+)|}) name with
```

Comment:
Please put the compilation part at toplevel !

Reply:
Ah yeah I meant to do that before committing :upside_down_face: